.github/actions/chart-test-recipes/action.yml

```diff
@@ -0,0 +1,179 @@
+    required: false
+    default: 'false'
+  flow:
+    description: "Test scenario to perform"
+    required: false
+    default: 'install'
+  extra-values:
+    description: 'Pass extra values to the Helm chart.'
+    required: false
+  ingress-hostname-base:
+    description: "Hostname base for the tests"
+    required: true
+  gh_app_id:
+    required: true
+    description: "Github App Id used to impersonate the action"
+  gh_app_private_key:
+    required: true
+    description: |
+      Github App private key used to impersonate the action.
+      Used to create/delete GitHub environment.
+      NOTE: The GH app requires "administration:write" access to be able to delete the GH environment.
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Generate GitHub token
+      uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2
+      id: generate-github-token
+      with:
+        app_id: ${{ inputs.gh_app_id }}
+        private_key: ${{ inputs.gh_app_private_key }}
+
+    - name: Set workflow vars
+      id: vars
+      uses: ./.github/actions/workflow-vars
+      with:
+        persistent: ${{ inputs.persistent }}
+        setup-flow: ${{ inputs.flow }}
+        platform: ${{ inputs.platform }}
+        type: ${{ inputs.type }}
+        identifier-base: ${{ inputs.identifier }}
+        ingress-hostname-base: ${{ inputs.ingress-hostname-base }}
+
+    - name: Install dependencies
+      uses: asdf-vm/actions/install@05e0d2ed97b598bfce82fd30daf324ae0c4570e6 # v3
+
+    - name: Add Helm repos and dependencies
+      shell: bash
+      run: |
+        make helm.repos-add
+        make helm.dependency-update
+
+    - name: Create test namespace
+      shell: bash
+      run: |
+        kubectl delete ns --ignore-not-found=true \
+          -l "github-id=${{ steps.vars.outputs.identifier }},test-flow=${{ inputs.flow }},test-persistent=true"
+        kubectl create ns $TEST_NAMESPACE
+        kubectl label ns $TEST_NAMESPACE github-run-id=$GITHUB_WORKFLOW_RUN_ID
+        kubectl label ns $TEST_NAMESPACE github-job-id=$GITHUB_WORKFLOW_JOB_ID
+        kubectl label ns $TEST_NAMESPACE github-id=${{ inputs.identifier }}
+        kubectl label ns $TEST_NAMESPACE test-flow=${{ inputs.flow }}
+        kubectl label ns $TEST_NAMESPACE test-persistent=${{ inputs.persistent }}
+
+    - name: Copy PRs wildcard certificate
+      shell: bash
+      run: kubectl apply -n $TEST_NAMESPACE -f .github/config/external-secret.yaml
+
+    - name: Start GitHub deployment
+      uses: bobheadxi/deployments@648679e8e4915b27893bd7dbc35cb504dc915bc8 # v1
+      id: deployment
+      with:
+        step: start
+        token: ${{ steps.generate-github-token.outputs.token }}
+        env: ${{ steps.vars.outputs.identifier }}
+        ref: ${{ inputs.caller-git-ref }}
+
+    - name: Pre setup
+      shell: bash
+      env:
+        TEST_CHART_FLOW: ${{ inputs.flow }}
+        TEST_INGRESS_HOST: ${{ steps.vars.outputs.ingress-host }}
```

Comment:
isn't `TEST_INGRESS_HOST` defined and exported as env as part of your `Set workflow vars` step already?